File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/dataholders/attributeInfos/AppleDeveloperCertificateAttribute.kt
package com.jetbrains.signatureverifier.serialization.dataholders

import com.jetbrains.signatureverifier.serialization.toPrimitiveDLSet
import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.cms.Attribute

@Serializable
data class AppleDeveloperCertificateAttribute(
  override val identifier: TextualInfo,
  val content: List<TextualInfo>
) : AttributeInfo() {

  constructor(attribute: Attribute) : this(
    TextualInfo.getInstance(attribute.attrType),
    attribute.attributeValues.map {
      TextualInfo.getInstance(it)
    }
  )

  override fun getPrimitiveContent() = content.toPrimitiveDLSet()
}

File: net/JetBrains.Serialization/src/Dataholders/AttributeInfos/AppleDeveloperCertificateAttribute.cs
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;
using Attribute = Org.BouncyCastle.Asn1.Cms.Attribute;
using JetBrains.Serialization;

[JsonObject(MemberSerialization.OptIn)]
public class AppleDeveloperCertificateAttribute : AttributeInfo
{
  [JsonProperty("Identifier")] protected override TextualInfo Identifier { get; }

  [JsonProperty("Content")] private List<TextualInfo> _content;

  [JsonConstructor]
  public AppleDeveloperCertificateAttribute(TextualInfo identifier, List<TextualInfo> content)
  {
    Identifier = identifier;
    _content = content;
  }

  public AppleDeveloperCertificateAttribute(Attribute attribute)
  {
    Identifier = TextualInfo.GetInstance(attribute.AttrType);
    _content = attribute.AttrValues.ToArray().Select(TextualInfo.GetInstance).ToList();
  }

  public override Asn1Encodable GetPrimitiveContent() => _content.ToPrimitiveDerSet();
}